torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -0,0 +1,296 @@
+import dataclasses
+import math
+from typing import Any, Dict, Tuple, Optional, Callable, List, cast, Iterator
+
+import torch
+from torchvision.prototype.transforms import Transform, InterpolationMode
+from torchvision.prototype.utils._internal import apply_recursively
+from torchvision.transforms import AutoAugment as _AutoAugment, AutoAugmentPolicy
+
+from . import functional as F
+from .functional._utils import Dispatcher
+from .utils import Query
+
+
+@dataclasses.dataclass
+class AutoAugmentDispatcher:
```

Comment:
Ad discussed offline we should consider removing this method and use lambdas on the `_DISPATCHER_MAP` or restore the old `_apply_op` method.